src/SFA.DAS.Funding.SystemAcceptanceTests/Helpers/Events/ApprenticeshipEventHelper.cs

```diff
@@ -0,0 +1,113 @@
+﻿using SFA.DAS.Funding.SystemAcceptanceTests.Hooks;
+using SFA.DAS.Funding.SystemAcceptanceTests.TestSupport;
+using SFA.DAS.CommitmentsV2.Types;
+using APR = SFA.DAS.Apprenticeships.Types;
+using CMT = SFA.DAS.CommitmentsV2.Messages.Events;
+
+namespace SFA.DAS.Funding.SystemAcceptanceTests.Helpers.Events;
+
+internal static class ApprenticeshipEventHelper
+{
+    internal static CMT.ApprenticeshipCreatedEvent CreateApprenticeshipCreatedMessageWithCustomValues(DateTime actualStartDate, DateTime plannedEndDate, decimal agreedPrice, string trainingCode)
+    {
+        var fixture = new Fixture();
+        return fixture.Build<CMT.ApprenticeshipCreatedEvent>()
+           .With(_ => _.StartDate, new DateTime(actualStartDate.Year, actualStartDate.Month, 1))
+           .With(_ => _.ActualStartDate, actualStartDate)
+           .With(_ => _.EndDate, plannedEndDate)
+           .With(_ => _.PriceEpisodes, new PriceEpisodeHelper().CreateSinglePriceEpisodeUsingStartDate(actualStartDate, agreedPrice))
+           .With(_ => _.Uln, GenerateRandomUln())
+           .With(_ => _.TrainingCode, trainingCode)
+           .With(_ => _.ApprenticeshipEmployerTypeOnApproval, ApprenticeshipEmployerType.Levy)
+           .With(_ => _.AccountId, 3871)
+           .With(_ => _.TransferSenderId, (long?)null)
+           .With(_ => _.DateOfBirth, DateTime.Now.AddYears((-25)))
+           .With(_ => _.IsOnFlexiPaymentPilot, true)
+           .With(_ => _.TrainingCourseVersion, "1.0")
+           .With(_ => _.ProviderId, 88888888)
+           .Create();
+    }
+
+    private static String GenerateRandomUln()
+    {
+        String randomUln = GenerateRandomNumberBetweenTwoValues(10, 99).ToString()
+            + DateTime.Now.ToString("ssffffff");
+
+        for (int i = 1; i < 30; i++)
+        {
+            if (IsValidCheckSum(randomUln))
+            {
+                return randomUln;
+            }
+            randomUln = (long.Parse(randomUln) + 1).ToString();
+        }
+        throw new Exception("Unable to generate ULN");
+    }
+
+    private static int GenerateRandomNumberBetweenTwoValues(int min, int max) => new Random().Next(min, max);
```

Comment:
newing up a Random each time like this might result in the same value being returned if it's called multiple times in quick succession because of how the seeding from DateTime.Now works, might be worth having an instance of Random as a static field.